Stratis.Bitcoin/BlockPulling/LookaheadBlockPuller.cs

```diff
@@ -385,38 +428,46 @@ private Block NextBlockCore(CancellationToken cancellationToken)
                 {
                     if (header.Previous.HashBlock != this.location.HashBlock)
                     {
-                        //A reorg
-                        return null;
+                        this.logger.LogTrace("Blockchain reorganization detected.");
+                        break;
                     }
                     this.IsStalling = false;
                     this.location = header;
                     Interlocked.Add(ref this.currentSize, -block.Length);
                     this.consumed.Set();
-                    return block.Block;
+
+                    res = block.Block;
                 }
                 else
                 {
-                    // Otherwise we either have reorg.
+                    // Otherwise we either have reorg, or we don't know the next block.
```

Comment:
Well we still consider chain as the tip, or at least the tip we know about, in reality if we are in IBD it means we didn't connect to a peer with a longer tip (yet).

As the chain never stops you may be right that we don't know the next block but that's because it wasn't mined yet.